app/src/org/commcare/views/widgets/SelectOneAutoAdvanceWidget.java

```diff
@@ -122,17 +123,30 @@ public SelectOneAutoAdvanceWidget(Context context, FormEntryPrompt prompt) {
 
                 addView(thisParentLayout);
             }
+            addClearButton(context, s != null && !prompt.isReadOnly());
         }
     }
 
+    private void addClearButton(Context context, boolean show) {
```

Comment:
this code is exactly same as the `addClearButton` in `SelectWidget` and should be reused.

Reply:
Yeah, would it be okay to have this method in `QuestionWidget`?

Reply:
you can create a new abstract class like `SelectQuestionWidget : QuestionWidget` to put the logic specific to selects.